src/Tizen.Applications.ComponentBased/Tizen.Applications.ComponentBased.Common/BaseType.cs

```diff
@@ -0,0 +1,173 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Tizen.Applications.CoreBackend;
+using static Interop.CBApplication;
+using static Tizen.Applications.CoreBackend.DefaultCoreBackend;
+
+namespace Tizen.Applications.ComponentBased.Common
+{
+    internal class BaseType
+    {
+        internal readonly Type _classType;
+        internal readonly string _compId;
+        internal IList<BaseComponent> _compInstances = new List<BaseComponent>();
+        private Interop.CBApplication.BaseLifecycleCallbacks _callbacks;
+        private BaseComponent.ComponentType _compType;
+        private CBApplicationBase _parent;
+
+        internal BaseType(Type ctype, string id, BaseComponent.ComponentType comp_type, CBApplicationBase parent)
```

Comment:
What is the responsibility of this class?
Implementation and naming is not matched

Implementations looks like a data of some runtime instance information
But, naming of `Type` is a static information. 

and I don't know why need it, it should be a part of BaseComponent

Reply:
This is an Abstract Factory pattern.
BaseType is responsible for creating base-component instances.
And BaseComponent class is a class that defines the behavior of base-component instance.  
By dividing these two behaviors, it is easier to maintain.
But, the name of this class is confusing. Therefore, let's change a name into BaseFactory.